### PR TITLE
add ovn DVR fip e2e

### PR DESCRIPTION
**Pull request by bobz965**
- [ ] Make sure you have followed [Kube-OVN Code Style](https://github.com/kubeovn/kube-ovn/blob/master/CODE_STYLE.md).

### What type of this PR
Examples of user facing changes:
- Tests

### Which issue(s) this PR fixes:
Fixes #(issue-number)

### <samp>🤖 Generated by Copilot at 3bb3282</samp>

This pull request fixes some typos, adds and updates e2e tests for the ovn-vpc-nat-gw feature, and changes the naming and formatting of the localnet port and name in various files. The purpose of these changes is to improve the testing, validation, and consistency of the ovn-vpc-nat-gw feature and the localnet port handling.

### <samp>🤖 Generated by Copilot at 3bb3282</samp>

> _We are the masters of the localnet port_
> _We set the tag and avoid the error report_
> _We change the prefix and the format of the name_
> _We test the gateway and unleash the NAT flame_

### <samp>🤖 Generated by Copilot at 3bb3282</samp>

*  Modify the prefix and format of the localnet logical switch port name to avoid conflicts and improve consistency ([link](https://github.com/kubeovn/kube-ovn/pull/2780/files?diff=unified&w=0#diff-2cdd0dec2c1ac53cc940998efdf213a7204dd26d6e5e8afd478ae8bcdf630e0eL30-R30), [link](https://github.com/kubeovn/kube-ovn/pull/2780/files?diff=unified&w=0#diff-2a7a7a33b9076c02e04dd02dcab8a68ac67159f3c233a75af4cd34917f52c055L22-R22), [link](https://github.com/kubeovn/kube-ovn/pull/2780/files?diff=unified&w=0#diff-8161e77fe331f5dd6eb9ecd97d22bc4bd88d5c967bdf8b0fa28478353434dfc3L23-R23))
* Add a check for the existence of the localnet port before setting its tag in the `setLocalnetTag` function ([link](https://github.com/kubeovn/kube-ovn/pull/2780/files?diff=unified&w=0#diff-7405fc764106a56c3d8c53ca59401ba6b0c285e3a136518b6be936880cbcd8d0R310-R313))
* Fix typos in the comments of the `ExpectNoErrorWithOffset`, `ExpectZero`, and `ExpectUUID` functions in `test/e2e/framework/expect.go` ([link](https://github.com/kubeovn/kube-ovn/pull/2780/files?diff=unified&w=0#diff-19f37de624b324e650f830dc887c4a5bba1831378790c62b09aee69a4ba0c2e9L70-R70), [link](https://github.com/kubeovn/kube-ovn/pull/2780/files?diff=unified&w=0#diff-19f37de624b324e650f830dc887c4a5bba1831378790c62b09aee69a4ba0c2e9L151-R151), [link](https://github.com/kubeovn/kube-ovn/pull/2780/files?diff=unified&w=0#diff-19f37de624b324e650f830dc887c4a5bba1831378790c62b09aee69a4ba0c2e9L161-R161))
* Add and update variables, ginkgo.By statements, and logic for the e2e test of the ovn-vpc-nat-gw feature in `test/e2e/ovn-vpc-nat-gw/e2e_test.go` ([link](https://github.com/kubeovn/kube-ovn/pull/2780/files?diff=unified&w=0#diff-c1831d6aa37f96132386404bff7ef6c738c40bcef66ba7492aca28929d61dca2R15), [link](https://github.com/kubeovn/kube-ovn/pull/2780/files?diff=unified&w=0#diff-c1831d6aa37f96132386404bff7ef6c738c40bcef66ba7492aca28929d61dca2L88-R90), [link](https://github.com/kubeovn/kube-ovn/pull/2780/files?diff=unified&w=0#diff-c1831d6aa37f96132386404bff7ef6c738c40bcef66ba7492aca28929d61dca2R96-R97), [link](https://github.com/kubeovn/kube-ovn/pull/2780/files?diff=unified&w=0#diff-c1831d6aa37f96132386404bff7ef6c738c40bcef66ba7492aca28929d61dca2R109), [link](https://github.com/kubeovn/kube-ovn/pull/2780/files?diff=unified&w=0#diff-c1831d6aa37f96132386404bff7ef6c738c40bcef66ba7492aca28929d61dca2L110-R121), [link](https://github.com/kubeovn/kube-ovn/pull/2780/files?diff=unified&w=0#diff-c1831d6aa37f96132386404bff7ef6c738c40bcef66ba7492aca28929d61dca2R276-R279), [link](https://github.com/kubeovn/kube-ovn/pull/2780/files?diff=unified&w=0#diff-c1831d6aa37f96132386404bff7ef6c738c40bcef66ba7492aca28929d61dca2L299-R311), [link](https://github.com/kubeovn/kube-ovn/pull/2780/files?diff=unified&w=0#diff-c1831d6aa37f96132386404bff7ef6c738c40bcef66ba7492aca28929d61dca2L340-R352), [link](https://github.com/kubeovn/kube-ovn/pull/2780/files?diff=unified&w=0#diff-c1831d6aa37f96132386404bff7ef6c738c40bcef66ba7492aca28929d61dca2L367-R379), [link](https://github.com/kubeovn/kube-ovn/pull/2780/files?diff=unified&w=0#diff-c1831d6aa37f96132386404bff7ef6c738c40bcef66ba7492aca28929d61dca2L379-R416), [link](https://github.com/kubeovn/kube-ovn/pull/2780/files?diff=unified&w=0#diff-c1831d6aa37f96132386404bff7ef6c738c40bcef66ba7492aca28929d61dca2R434), [link](https://github.com/kubeovn/kube-ovn/pull/2780/files?diff=unified&w=0#diff-c1831d6aa37f96132386404bff7ef6c738c40bcef66ba7492aca28929d61dca2L428-R443), [link](https://github.com/kubeovn/kube-ovn/pull/2780/files?diff=unified&w=0#diff-c1831d6aa37f96132386404bff7ef6c738c40bcef66ba7492aca28929d61dca2L448-R523), [link](https://github.com/kubeovn/kube-ovn/pull/2780/files?diff=unified&w=0#diff-c1831d6aa37f96132386404bff7ef6c738c40bcef66ba7492aca28929d61dca2R530-R533))

### CODE REVIEW

File: pkg/ovs/util.go [hunk 1]
```diff
@@ -20,7 +20,7 @@ func PodNameToPortName(pod, namespace, provider string) string {
 }
 
 func GetLocalnetName(subnet string) string {
-	return fmt.Sprintf("localnet.%s", subnet)
+	return fmt.Sprintf("localnet-%s", subnet)
 }
```
Comment (zhangzujian) [inline pkg/ovs/util.go:24]: Why change localnet LSP name?

Comment (bobz965) [reply pkg/ovs/util.go:24]: LocalnetName fmt should be consistent

File: pkg/controller/vlan.go [hunk 2]
```diff
@@ -307,6 +307,10 @@ func (c *Controller) updateProviderNetworkStatusForVlanDeletion(pn *kubeovnv1.Pr
 
 func (c *Controller) setLocalnetTag(subnet string, vlanID int) error {
 	localnetPort := ovs.GetLocalnetName(subnet)
+	if _, err := c.ovnClient.GetLogicalSwitchPort(localnetPort, false); err != nil {
+		klog.Errorf("wait for localnet port %s to be exist: %v", localnetPort, err)
+		return err
+	}
```
Comment (zhangzujian) [inline pkg/controller/vlan.go:313]: No need  to check LSP existence here.

Comment (bobz965) [reply pkg/controller/vlan.go]: ok

**Review by zhangzujian — CHANGES_REQUESTED** [review]

File: test/e2e/ovn-vpc-nat-gw/e2e_test.go [hunk 3]
```diff
@@ -445,19 +459,82 @@ var _ = framework.Describe("[group:ovn-vpc-nat-gw]", func() {
 		ginkgo.By("Deleting ovn vip " + dnatVipName)
 		vipClient.DeleteSync(dnatVipName)
 
-		ginkgo.By("Deleting custom vpc " + vpcName)
-		vpcClient.DeleteSync(vpcName)
+		ginkgo.By("Creating config map ovn-external-gw-config for distributed case")
+		cmData = map[string]string{
+			"enable-external-gw": "true",
+			"external-gw-nodes":  "kube-ovn-control-plane,kube-ovn-worker",
```
Comment (zhangzujian) [inline test/e2e/ovn-vpc-nat-gw/e2e_test.go:465]: Generate gateway nodes from node nodes in the current k8s cluster.

File: test/e2e/ovn-vpc-nat-gw/e2e_test.go [hunk 4]
```diff
@@ -445,19 +459,82 @@ var _ = framework.Describe("[group:ovn-vpc-nat-gw]", func() {
...
+			"type":               "distributed",
+			"external-gw-nic":    "eth1",
+			"external-gw-addr":   strings.Join(cidr, ","),
+		}
+		// TODO:// external-gw-nodes could be auto managed by recognizing gw chassis node which has the external-gw-nic
+		configMap = &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "ovn-external-gw-config",
```
Comment (zhangzujian) [inline test/e2e/ovn-vpc-nat-gw/e2e_test.go:473]:
```suggestion
				Name:      util.ExternalGatewayConfig,
```

File: test/e2e/ovn-vpc-nat-gw/e2e_test.go [hunk 5]
```diff
@@ -445,19 +459,82 @@ var _ = framework.Describe("[group:ovn-vpc-nat-gw]", func() {
...
 
 		k8sNodes, err = e2enode.GetReadySchedulableNodes(cs)
 		framework.ExpectNoError(err)
 		for _, node := range k8sNodes.Items {
 			// label should be false after remove node external gw
 			framework.ExpectHaveKeyWithValue(node.Labels, util.NodeExtGwLabel, "false")
 		}
+
+		ginkgo.By("Deleting configmap")
+		err = cs.CoreV1().ConfigMaps(framework.KubeOvnNamespace).Delete(context.Background(), "ovn-external-gw-config", metav1.DeleteOptions{})
```
Comment (zhangzujian) [inline test/e2e/ovn-vpc-nat-gw/e2e_test.go:536]:
```suggestion
		err = cs.CoreV1().ConfigMaps(configMap.Namespace).Delete(context.Background(), configMap.Name, metav1.DeleteOptions{})
```

**Comment by zhangzujian** [comment]
The e2e test still fails occasionally. We should fix it.

**Comment by bobz965** [comment]
> The e2e test still fails occasionally. We should fix it.

ok, on the way

**Review by zhangzujian — APPROVED** [review]

**Pull request merged**